### PR TITLE
Update ARP to display incomplete entries

### PR DESCRIPTION
**Pull request by bogdanPricope**
Changes:
- locate correct entry for saved pkts
- show incomplete entries
- use 'is_complete' flag instead of 'if_valid' where that was intended 
- replace 'is_valid' field with 'is_used' flag

### CODE REVIEW

File: src/ofp_arp.c [hunk 1]
```diff
@@ -608,13 +611,23 @@ enum ofp_return_code ofp_arp_save_ipv4_pkt(odp_packet_t pkt, struct ofp_nh_entry
 
 	odp_rwlock_write_lock(&shm->arp.set[set].table_rwlock);
 
-#if (ARP_SANITY_CHECK)
-	newarp = arp_lookup(set, &key);
-       if (newarp && newarp->is_valid)
-		OFP_ERR("ARP Entry failed the sanity check!");
-#endif
+	if (is_link_local) {
+		odp_rwlock_t *lock = &shm->arp.set[set].table_rwlock;
```
Comment (matiaselo) [inline src/ofp_arp.c:614]: `odp_rwlock_t *lock` could be moved to upper level so all lock/unlock functions could use the same variable making the code a bit easier to read.

File: src/ofp_arp.c [hunk 2]
```diff
@@ -708,7 +721,9 @@ void ofp_arp_show_table(int fd)
 	for (i = 1; i < NUM_ARPS; ++i) {
 		entry = &shm->arp.entries[i];
 
-		if (entry->flags.is_complete)
+		if (entry->flags.is_complete ||
+		    (!entry->flags.is_complete &&
```
Comment (matiaselo) [inline src/ofp_arp.c:722]: No need to repeat this clause after the first one.

**Comment by matiaselo** [comment]
I added couple minor comments and there is a typo in the commit log of `arp: fix the presentation of incomplete entries`: asociated -> associated. Otherwise the is PR looking good.

File: src/ofp_arp.c (hunk 2 above)
Comment (bogdanPricope) [reply src/ofp_arp.c:722]: OK

File: src/ofp_arp.c (hunk 1 above)
Comment (bogdanPricope) [reply src/ofp_arp.c:614]: OK

**Pull request merged**